pkg/operator/csi/csidrivercontrollerservicecontroller/csi_driver_controller_service_controller.go

```diff
@@ -77,19 +87,28 @@ func NewCSIDriverControllerServiceController(
 	operatorClient v1helpers.OperatorClient,
 	kubeClient kubernetes.Interface,
 	deployInformer appsinformersv1.DeploymentInformer,
+	configInformers configinformers.SharedInformerFactory,
 	recorder events.Recorder,
 ) factory.Controller {
 	c := &CSIDriverControllerServiceController{
-		name:           name,
-		manifest:       manifest,
-		operatorClient: operatorClient,
-		kubeClient:     kubeClient,
-		deployInformer: deployInformer,
+		name:            name,
+		manifest:        manifest,
+		operatorClient:  operatorClient,
+		kubeClient:      kubeClient,
+		deployInformer:  deployInformer,
+		configInformers: configInformers,
 	}
 
-	return factory.New().WithInformers(
+	informers := []factory.Informer{
 		operatorClient.Informer(),
 		deployInformer.Informer(),
+	}
+	if c.configInformers != nil {
```

Comment:
why should this ever be nil? Feels like unneeded complexity.

Reply:
Some CSI drivers (operands) don't have the option to pass in a cluster ID, so their operators will let this to be nil

Reply:
Also, the docstring above tries to explain that

Reply:
call it optionalConfigInformer. The current pattern is a recipe to null pointer exceptions the next time somebody modifies this code and is not aware of this invariant.

Reply:
Renamed to `optionalConfigInformer`